lib/DocumentManager.php

```diff
@@ -175,4 +265,34 @@ private function getOptionsResolver($eventName)
 
         return $resolver;
     }
+
+    /**
+     * If the exception is an instanceof `DocumentManagerException` set the
+     * document manager name that threw.
+     *
+     * @param \Exception $e
+     * @param string $context
+     */
+    private function processException(\Exception $exception, $context)
+    {
+        $message = sprintf('Error %s', $context);
+
+        if ($exception instanceof DocumentManagerException) {
+            // if the exception already has a document manager name, then it
+            // was thrown by this method in another document manager instance
+            // so we wrap it again.
+            //
+            // the user-facing exception class will be lost and we are assuming that standard
+            // exceptions (`DocumentNotFoundException`) will not be affected by this
+            // within the scope of normal usage.
+            if ($exception->getDocumentManagerName()) {
+                throw new DocumentManagerException($message, $this->context->getName(), $exception);
```

Comment:
Are you sure that's right? When the live DM is called last, it will add its own name to the exception. From what I can spot here it's exactly the opposite from what you are describing.

1. You save a document in default DM.
2. The synchronizatino subscriber saves to the live DM.
3. Live DM throws an exception
4. Live DM catches the exception and sets the name
5. Default DM also catches the Exception and sets the name again.

So, as it is currently implemented, you are ending up with the default name, although it has been failing in the live document manager.